### ISSUE TITLE
Cannot create azurerm_redis_enterprise_cluster in North Central US region

### ISSUE THREAD

**Issue by OlehZarevych** [hashicorp/terraform-provider-azurerm#22640]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

v1.5.3

### AzureRM Provider Version

v3.55.0

### Affected Resource(s)/Data Source(s)

azurerm_redis_enterprise_cluster 

### Terraform Configuration Files

```hcl
resource "azurerm_redis_enterprise_cluster" "example" {
  name                = "example-redisenterprise"
  resource_group_name = azurerm_resource_group.example.name
  location            = "North Central US"
  sku_name = "EnterpriseFlash_F300-3"
}
```

### Debug Output/Panic Output

```shell
Error: "north central us" does not currently support Redis Enterprise Clusters. Locations which currently support Redis Enterprise Clusters are ["Australia East", "Australia Southeast", "Brazil South", "Canada Central", "Central India", "Central US", "Central US EUAP", "East Asia", "East US", "North Europe", "South Central US", "South India", "Southeast Asia", "UK South", "UK West", "East US 2", "East US 2 EUAP", "West Europe", "West US" or "West US 2"]
```

### Expected Behaviour

We can create resource in North Central US

### Actual Behaviour

Error :

Error: "north central us" does not currently support Redis Enterprise Clusters. Locations which currently support Redis Enterprise Clusters are ["Australia East", "Australia Southeast", "Brazil South", "Canada Central", "Central India", "Central US", "Central US EUAP", "East Asia", "East US", "North Europe", "South Central US", "South India", "Southeast Asia", "UK South", "UK West", "East US 2", "East US 2 EUAP", "West Europe", "West US" or "West US 2"]

### Steps to Reproduce

1. terraform apply

**Comment by xuzhang3** [comment]
@OlehZarevych this issue has been fixed in #22578

**Comment by OlehZarevych** [comment]
@xuzhang3 thanks ! Will close this ticket !
